### PR TITLE
Add a new method getValidFrom

### PR DESCRIPTION
**Pull request by bilfeldt**
Add a convenience method `getValidFrom` to the OneTimePassword class

### CODE REVIEW

File: tests/OneTimePasswordTest.php [hunk 1]
```diff
@@ -17,3 +17,14 @@
     expect($dto->getValidUntil())->toBe($validUntil);
     expect($dto->getValidityTimespan())->toBe($validityTimespan);
 });
+
+test('get valid from calculated correctly', function () {
```
Comment (comes) [inline tests/OneTimePasswordTest.php:21]: this test is wrong.

OTPs are only valid between seconds `0` and `29` or `30` to `59`. Every half minute, a new token needs to be calculated. In your test, the getValidFrom() has to be `2022, 8, 22, 13, 14, 30`.

for that reason, it is absolutely not necessary to have a ValidFrom sitting there. Everyone knows, that this token can only be created in second `0` or `30`. I would recommend to delete this part.

**Review by comes — CHANGES_REQUESTED** [review]
a validFrom is not needed. The specification says, token are always generated at second 0 or second 30. never in between.

**Comment by bilfeldt** [comment]
The logic here is clearely flawed and should be fixed before merging.

The motivation for this PR is to add enough data to the `OneTimePassword` object to both show the password, but also a piechart of "percentage of lifetime remaining" - much like the Google Authenticator does it:
2FA-iPhone-946x2048

Without this method, then the logic for calculating this "percentage" would not be provided by this package.

**Comment by comes** [comment]
I get the motivation and understand the usecase. IMO this can be easily implemented in the frontend already

```blade
<div class="bg-white overflow-hidden shadow rounded-lg" x-data="{
  until: new Date({{ $otp->getValidUntil()->getTimestampMs() }}),
  isExpired: false,
  percentage: 0,
  validityTimespan: {{ $otp->getValidityTimespan()->totalSeconds * 1000 }},
  interval: null,
  tick() {
      let timeDiff = this.until - Date.now();
      this.percentage = 1 - (timeDiff / this.validityTimespan);
      console.log({percentage: this.percentage, msg : 'tick-tock'});
  },
  init() {
      this.interval = setInterval(() => this.tick(), 1000)
  },
}">
	<span x-text="percentage"></span>
</div>
```

**Comment by bilfeldt** [comment]
Imagine you generate a `OneTimePassword` and wanna show it like Google Authenticator. Then you need to know:
- Password
- Percentage (for the piechart)
- Seconds left (if you wanna show a count down)

Percentage is simply calculated as: `(validTo - now) / window` so we do not need this function after all 🤦

**Pull request closed without merging**